src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        growIfArrayFull();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        growIfArrayFull();
+        for (int i = 0; i < list.size(); i++) {
+            elements[size] = list.get(i);
+            size++;
```

Comment:
It should be better to call your existing method `add()`.
```suggestion
            add(list.get(i));
```

You are resizing the array only once before the loop. What will happen, if array will become full, but loop is still processing some number of elements? Your method `add` already contains the `growIfArrayFull` logic.